pkg/kubectl/cmd/cp/cp.go

```diff
@@ -275,11 +275,42 @@ func (o *CopyOptions) copyToPod(src, dest fileSpec, options *exec.ExecOptions) e
 	return o.execute(options)
 }
 
+// Changes the destination filename and location if they are not explicitly specified.
+// If destination is a directory, filename will be the same as the source.
+// Also, symlink destinations are converted to true paths
+func convertHostDestFilename(srcFile string, destFile string) (string, error) {
+	stat, err := os.Stat(destFile)
+	if err != nil && !os.IsNotExist(err) {
+		return "", err
+	}
+
+	if !os.IsNotExist(err) && stat.IsDir() {
```

Comment:
Tar will append the right filename.
No need to do like this.

Reply:
I needed to do this to address the `kubectl cp` bug that occurs now when you use to on a directory. That is why I need to do this.

Reply:
@pswica Actually this is a bug of wrong `prefix` when untar. We do have a file prefix checking there.

In https://github.com/kubernetes/kubernetes/pull/78928/files#diff-5103d5139d8d0e3143b6d7b6d1934f50R339, I've included a fix for this issue.

Reply:
Interesting. I guess it depends where you think the logic should go in implementing the path that gets put into tar. I prefer my approach as it cleans everything before sending it to tar. This, to me, separates the logic between the two functions and makes them more loosely coupled . 

Perhaps @liggitt  or @siosphere can weigh in here.

Reply:
@dixudx another thing I'll mention, looking at your PR, is that it currently has a lot of failing tests:

```
I0616 14:51:04.155] should copy a file from a running Pod [It]
I0616 14:51:24.751] should copy a file from a running Pod [It]
I0616 14:53:57.520] runs ReplicaSets to verify preemption running path [It]
...
```

I don't think it would be fair to state that, at this moment, your solution fixes any bugs that mine fixes.